### PR TITLE
Javadocs for interfaces and abstract calsses

### PR DESCRIPTION
**Pull request by oubaydos**

### CODE REVIEW

File: src/main/java/io/javarig/generator/TypeBasedGenerator.java [hunk 1]
```diff
@@ -2,8 +2,12 @@
 
 import java.lang.reflect.Type;
 
+/**
+ * type generator for classes that require an extra type
```
Comment (HamzaBenyazid) [inline src/main/java/io/javarig/generator/TypeBasedGenerator.java]: This is not clear 
I think better say :  type generator for classes that require the Type object that holds information about the object to generate

File: src/main/java/io/javarig/generator/list/ListGenerator.java [hunk 2]
```diff
@@ -13,6 +13,11 @@
 import java.lang.reflect.Type;
 import java.util.List;
 
+/**
+ * a type generator that generates a list instance,
+ * <p>
+ * note: generates an ArrayList if not given otherwise
```
Comment (HamzaBenyazid) [inline src/main/java/io/javarig/generator/list/ListGenerator.java]: This information is not handled here. The code doesn't generate ArrayList if no implementation type was given. We did that in the TypeEnum when we gave the LIST enum ArrayListGenerator.
I find this comment confusing.

File: src/main/java/io/javarig/generator/map/MapGenerator.java [hunk 3]
```diff
@@ -11,6 +11,11 @@
 import java.lang.reflect.Type;
 import java.util.Map;
 
+/**
+ * a type generator that generates a map instance,
+ * <p>
+ * Note: generates a hashmap if not given otherwise
```
Comment (HamzaBenyazid) [inline src/main/java/io/javarig/generator/map/MapGenerator.java]: The same applies here

**Review by HamzaBenyazid — CHANGES_REQUESTED** [review]

File: src/main/java/io/javarig/generator/list/ListGenerator.java (hunk 2 above)
Comment (HamzaBenyazid) [reply src/main/java/io/javarig/generator/list/ListGenerator.java]: check **Nonlocal Information** in chapter 4 of **Clean Code** book
https://thixalongmy.haugiang.gov.vn/media/1175/clean_code.pdf

**Review by mosle77 — APPROVED** [review]

**Review by HamzaBenyazid — APPROVED** [review]

**Pull request merged**